include/neml2/misc/types.h

```diff
@@ -65,6 +66,14 @@ static const torch::TensorOptions default_tensor_options = torch::TensorOptions(
                                                                .device(torch::kCPU)
                                                                .requires_grad(false);
 
+/// We similarly want to have a default integer scalar type for some types of tensors
+#define TORCH_INT_DTYPE CONCAT(TORCH_ENUM_PREFIX, INT_DTYPE)
+static const torch::TensorOptions default_integer_tensor_options = torch::TensorOptions()
+                                                                       .dtype(TORCH_INT_DTYPE)
+                                                                       .layout(torch::kStrided)
+                                                                       .device(torch::kCPU)
+                                                                       .requires_grad(false);
```

Comment:
```suggestion
static const torch::TensorOptions default_integer_tensor_options = default_tensor_options.dtype(TORCH_INT_DTYPE);
```

just in case we want to have non-default layout, device, requires_grad in the future.